boot/debug.go

```diff
@@ -83,3 +86,37 @@ func DumpBootVars(w io.Writer, dir string, uc20 bool) error {
 	}
 	return nil
 }
+
+// DebugSetBootVars is a debug helper that takes a list of <var>=<value> entries
+// and sets them for the configured bootloader.
+func DebugSetBootVars(dir string, varEqVal []string) error {
+	opts := &bootloader.Options{
+		NoSlashBoot: dir != "" && dir != "/",
+	}
+	switch dir {
+	// is it any of the well-known UC20 boot partition mount locations?
+	case InitramfsUbuntuBootDir:
+		opts.Role = bootloader.RoleRunMode
+	case InitramfsUbuntuSeedDir:
+		opts.Role = bootloader.RoleRecovery
+	}
+	if !opts.NoSlashBoot {
+		// no root directory set, default to run mode
+		opts.Role = bootloader.RoleRunMode
```

Comment:
how do we know it's RoleRunMode vs RoleSole here? are we only using DebugSetBootVars for uc20 systems? if not then this could be wrong for some bootloader implementations i.e. lk which has a specifically different behavior from run mode uc20 and normal mode uc18.

Reply:
also if this function is indeed only meant to be used for uc20 systems, then the function name should include 20 in the name

Reply:
I guess we need a --uc20 switch like for boot-vars